docs/src/modules/java/pages/serialization.adoc

```diff
@@ -7,40 +7,23 @@ You do not need to handle serialization for messages. Akka Serverless functions
 
 The gRPC services are also exposed as HTTP endpoints with JSON messages. See xref:proto.adoc#_transcoding_http[Transcoding HTTP].
 
-For messages consumed from or published to topics it can be needed to use another format than `protobuf`. Akka Serverless provides a utility to serialize and deserialize JSON messages.
+== When consuming Messages from Topics
 
-== Primitive types
+When a message arrives from a topic, Akka Serverless detects the message payload type based on the `Content-Type` or `ce-datacontenttype` header or attribute of the message. If there is no such metadata, the content is seen as raw bytes.
 
-Akka Serverless supports serializing the following primitive types:
+If the content type starts with `application/protobuf`, `application/x-protobuf` or `application/vnd.google.protobuf` the payload is expected to also have a `ce-type` header or attribute identifying the concrete protobuf message type. Such messages will be decoded into the described message type before handed to a topic subscriber method, which must accept that specific message type.
 
-[cols="a,a"]
-|===
-| Protobuf type | Java type
+If the publishing service is also an Akka Serverless service, this is handled transparently for you as shown in xref:actions-publishing-subscribing.adoc[].
 
-| string
-| java.lang.String
+For messages consumed from or published to topics to interact with external services it can be a requirement to use another format than `protobuf`.
```

Comment:
Just a suggested tweaking:

```suggestion
For messages that are consumed from or published to topics when interacting with external services, it can be a requirement to use a format other than `protobuf`. Other supported message formats include JSON, text, or raw bytes.
```